.github/workflows/ghcr_runtime.yml

```diff
@@ -141,14 +138,17 @@ jobs:
         run: make install-python-dependencies
       - name: Run runtime tests
         run: |
+          # We install pytest-xdist in order to run tests across CPUs
+          poetry run pip install pytest-xdist
+
           image_name=ghcr.io/${{ github.repository_owner }}/runtime:${{ github.sha }}-${{ matrix.base_image }}
           image_name=$(echo $image_name | tr '[:upper:]' '[:lower:]')
 
           TEST_RUNTIME=eventstream \
           SANDBOX_USER_ID=$(id -u) \
           SANDBOX_BASE_CONTAINER_IMAGE=$image_name \
           TEST_IN_CI=true \
-          poetry run pytest --cov=agenthub --cov=openhands --cov-report=xml -s ./tests/runtime
+          poetry run pytest -n 4 --cov=agenthub --cov=openhands --cov-report=xml -s ./tests/runtime
```

Comment:
Sonnet told me that since we are using code coverage (--cov), we might also want to use this flag:
`--concurrency=multiprocessing`
```
Coverage file handling:
When using --concurrency=multiprocessing, pytest-cov will create separate coverage files for each process and then combine them at the end of the test run.
```

Reply:
I will take a look. Have to figure out the failing tests though

Reply:
Looking at this. Are we not using pytest-cov? Seems like it should be able to handle it?
https://pytest-cov.readthedocs.io/en/latest/xdist.html

Reply:
Yes, I was only hinting about that extra option, but idk if it is making things better or is needed.

Reply:
Is there a way I can check the coverage for this and see if it lines up with what we are currently at?